state/syncer/userAccountsSyncer.go

```diff
@@ -119,27 +117,45 @@ func (u *userAccountsSyncer) SyncAccounts(rootHash []byte) error {
 
 	go u.printStatisticsAndUpdateMetrics(ctx)
 
-	mainTrie, err := u.syncMainTrie(rootHash, factory.AccountTrieNodesTopic, ctx)
-	if err != nil {
-		return err
+	leavesChannels := &common.TrieIteratorChannels{
+		LeavesChan: make(chan core.KeyValueHolder, common.TrieLeavesChannelDefaultCapacity),
+		ErrChan:    errChan.NewErrChanWrapper(),
 	}
 
-	defer func() {
-		_ = mainTrie.Close()
-	}()
+	wgSyncMainTrie := &sync.WaitGroup{}
+	wgSyncMainTrie.Add(1)
+
+	go func() {
+		err := u.syncMainTrie(rootHash, factory.AccountTrieNodesTopic, ctx, leavesChannels)
```

Comment:
I think you can pass only the `leavesChannels.LeavesChan` here, `ErrChan` is not used iin this func.

Reply:
refactored to use only leaves chan